### ISSUE TITLE
Suppress RedundantVisibilityModifierRule if explicit API mode enabled

### ISSUE THREAD

**Issue by osipxd** [detekt/detekt#3125]
## Expected Behavior of the rule

If explicit API mode is enabled:
```kotlin
kotlin {
    explicitApi()
}
```
Detekt should suppress `RedundantVisibilityModifierRule` for `public` modifier.

## Context

[Explicit API mode](https://kotlinlang.org/docs/reference/whatsnew14.html#explicit-api-mode-for-library-authors) was added in Kotlin 1.4
It prevents libraries' authors from making APIs public unintentionally. In this mode, you should explicitly specify the visibility modifier even if it is `public`.

**Comment by osipxd** [comment]
Looks like it is not fixed in 1.15.0
I've created the minimal project to reproduce this bug.
https://github.com/osipxd/detekt-explicit-api-mode-test

**Comment by osipxd** [comment]
Looks like the value in AnalysisFlags, used to get explicitMode is default value, so it is always `DISABLED` and `AnalysisFlag` is immutable.
https://github.com/JetBrains/kotlin/blob/8be23df668be32bad8c23f49f0b9f8feb7cb6cd9/compiler/util/src/org/jetbrains/kotlin/config/AnalysisFlag.kt#L32-L34

**Comment by schalkms** [comment]
Thanks for checking this feature thoroughly.
I scrolled through the [version history](
https://github.com/JetBrains/kotlin/commit/7058492b55841b33503131efee849524dade9c5f) of the Kotlin compiler. Luckily, I found another possible solution by using another internal function.
Please check out the referenced PR. Thank you!
